sdk/src/storage-settings.ts

```diff
@@ -173,7 +173,11 @@ export class StorageSettings<T extends string> implements Settings {
             }
             return setting.defaultValue;
         };
-        const ret = parseValue(this.device.storage.getItem(key), setting, readDefaultValue, rawDevice);
+        const item = this.device.storage.getItem(key);
+        if (!item) {
```

Comment:
not found is valid, it will use the default value or return undefined. should not be throwing.

Reply:
Right maybe in that case `parseValue` can be changed to accept a nullable `value`?

```ts
function parseValue(value: string, setting: StorageSetting, readDefaultValue: () => any, rawDevice?: boolean) {
```

becomes
```ts
function parseValue(value: string | null, setting: StorageSetting, readDefaultValue: () => any, rawDevice?: boolean) {
    if (value === null) {
        return readDefaultValue();
    }
    ...
```

Reply:
Or alternatively we can also inline in this function

```ts
        const item = this.device.storage.getItem(key);
        const ret = item !== null ? parseValue(item, setting, readDefaultValue, rawDevice) : readDefaultValue();
```

Reply:
@koush do you have a preference?

Reply:
parsing a value of null/undefined is valid, so the prior.